### PR TITLE
Feat (dev env): allow to custom registry port for `make kind-up`

### PR DESCRIPTION
**Pull request by zou8944**
### Type of change:

- [x] New feature provided

### What this PR does / why we need it:

https://github.com/apache/apisix-ingress-controller/issues/1416

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

**Review by tao12345666333 — APPROVED** [review]

File: utils/kind-with-registry.sh [hunk 1]
```diff
@@ -34,6 +34,9 @@ kind_version=$(kind version)
 kind_network='kind'
 reg_name='kind-registry'
 reg_port='5000'
+if [ -n "$1" ]; then
```
Comment (lingsamuel) [inline utils/kind-with-registry.sh:37]: I think `${1:-5000}` is ok?

Comment (zou8944) [reply utils/kind-with-registry.sh:37]: Sure, I will update it.

File: utils/kind-with-registry.sh [hunk 2]
```diff
@@ -67,7 +67,7 @@ nodes:
 containerdConfigPatches:
 - |-
   [plugins."io.containerd.grpc.v1.cri".registry.mirrors."localhost:${reg_port}"]
-    endpoint = ["http://${reg_host}:${reg_port}"]
+    endpoint = ["http://${reg_host}:5000"]
```
Comment (zou8944) [inline utils/kind-with-registry.sh:70]: Hi, @tao12345666333 @lingsamuel , I update this port, it should be fixed 5000 port.

**Comment by zou8944** [comment]
@tao12345666333 @lingsamuel , New commit updated, please review, thanks.

File: utils/kind-with-registry.sh (hunk 2 above)
Comment (tao12345666333) [reply utils/kind-with-registry.sh:70]: > Hi, @tao12345666333 @lingsamuel , I update this port, it should be fixed 5000 port.

why it should be 5000?

Comment (zou8944) [reply utils/kind-with-registry.sh:70]: @tao12345666333 the endpoint will be visit by k8s node which kind created, so the port should be the one that `registry` exposed (which is 5000 fixed), not the one that publish on our host. 
you can also refer to the [kind doc](https://kind.sigs.k8s.io/docs/user/local-registry/), and pay attention to this part
image

Comment (tao12345666333) [reply utils/kind-with-registry.sh:70]: okay, I miss it.

**Review by tao12345666333 — APPROVED** [review]

**Review by lingsamuel — APPROVED** [review]

**Review by lingsamuel — APPROVED** [review]

**Pull request merged**